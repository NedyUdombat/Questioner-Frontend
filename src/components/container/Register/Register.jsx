import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Loader, Dimmer } from 'semantic-ui-react';

// actions
import { register } from '../../../store/modules/auth';

// components
import Input from '../../presentational/Input/Input';
import Button from '../../presentational/Button/Button';

// styles
import './Register.scss';

export class Register extends React.Component {
  state = {
    userData: {
      email: '',
      password: '',
      username: '',
      firstname: '',
      lastname: '',
      phonenumber: '',
      othername: '',
      confirmpassword: '',
    },
  };

  userInputHandler = event => {
    this.setState({
      userData: {
        ...this.state.userData,
        [event.target.name]: event.target.value,
      },
    });
  };

  submitHandler = async event => {
    event.preventDefault();
    const { userData } = this.state;
    this.props.register({
      ...userData,
    });

    this.setState({
      userData: {
        ...userData,
        password: '',
      },
    });
  };

  render() {
    const { isLoading } = this.props;
    return (
      <div>
        <p
          className="f-50 text-center font-title"
          style={{
            marginBottom: '0.625rem',
          }}
        >
          Register
        </p>
        <div className="register-card">
          <form onSubmit={this.submitHandler}>
            {isLoading && (
              <Dimmer active>
                <Loader size="large">Loading</Loader>
              </Dimmer>
            )}
            <div className="form-group">
              <div className="input-div d-flex justify-content-center">
                <Input
                  onChange={this.userInputHandler}
                  className="form-control-input"
                  value={this.state.userData.firstname}
                  placeholder="First Name"
                  name="firstname"
                  type="text"
                />
              </div>
              <div className="input-div d-flex justify-content-center">
                <Input
                  onChange={this.userInputHandler}
                  className="form-control-input"
                  value={this.state.userData.lastname}
                  placeholder="Last Name"
                  name="lastname"
                  type="text"
                />
              </div>
              <div className="input-div d-flex justify-content-center">
                <Input
                  onChange={this.userInputHandler}
                  className="form-control-input"
                  value={this.state.userData.othername}
                  placeholder="Other Name"
                  name="othername"
                  type="text"
                />
              </div>
              <div className="input-div d-flex justify-content-center">
                <Input
                  onChange={this.userInputHandler}
                  className="form-control-input"
                  value={this.state.userData.username}
                  placeholder="Username"
                  name="username"
                  type="text"
                  required="required"
                />
              </div>
              <div className="input-div d-flex justify-content-center">
                <Input
                  onChange={this.userInputHandler}
                  className="form-control-input"
                  value={this.state.userData.email}
                  placeholder="Email"
                  name="email"
                  type="email"
                  required="required"
                />
              </div>
              <div className="input-div d-flex justify-content-center">
                <Input
                  onChange={this.userInputHandler}
                  className="form-control-input"
                  value={this.state.userData.phonenumber}
                  placeholder="Phone Number"
                  name="phonenumber"
                  type="tel"
                  pattern="{+0-9}[13]"
                />
              </div>
              <div className="input-div d-flex justify-content-center">
                <Input
                  onChange={this.userInputHandler}
                  className="form-control-input"
                  value={this.state.userData.password}
                  placeholder="Password"
                  name="password"
                  type="password"
                  pattern="(?=.*[a-z]).{6,}"
                  title="Must be 6 characters or more and contain at least lowercase letter"
                  required="required"
                />
              </div>
              <div className="input-div d-flex justify-content-center">
                <Input
                  onChange={this.userInputHandler}
                  className="form-control-input"
                  value={this.state.userData.confirmpassword}
                  placeholder="Confirm Password"
                  name="confirmpassword"
                  type="password"
                  required="required"
                />
              </div>
            </div>
            <div
              style={{
                marginTop: '.9235rem',
              }}
            >
              <Button type="submit" value="GET STARTED" className="btn-dark" />
            </div>
            <p
              className="f-20 text-grey text-center"
              style={{
                marginTop: '.9235rem',
              }}
            >
              Already have an account? Log in{' '}
              <Link to="/" className="text-blue">
                here
              </Link>
            </p>
          </form>
        </div>
      </div>
    );
  }
}

Register.propTypes = {
  register: PropTypes.func,
  isLoading: PropTypes.bool,
};

export const mapStateToProps = state => ({
  isLoading: state.auth.isLoading,
});

export default connect(
  mapStateToProps,
  { register },
)(Register);
