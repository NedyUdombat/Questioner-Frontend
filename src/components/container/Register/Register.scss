@import '../../../assets/sass/index';

.form-group {
  display: grid;
  grid-gap: .9375rem;
  grid-template-columns: auto auto;
  padding: 0;

  .input-div {
    font-size: 1.825rem;
    height: 3.75rem;
    padding: 0;
    text-align: center;
  }
}

@media only screen and (max-width: 47.5rem) {
  .form-group {
    grid-template-columns: auto;
  }
}
