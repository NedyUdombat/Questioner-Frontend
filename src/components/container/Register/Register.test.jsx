import React from 'react';
import { mount } from 'enzyme/build';
import { BrowserRouter } from 'react-router-dom';

import { Register, mapStateToProps } from './Register';

const props = {
  isLoading: false,
  register: jest.fn(),
};

const loadingProps = {
  isLoading: true,
  register: jest.fn(),
};

const mountWrapper = mount(
  <BrowserRouter>
    <Register {...props} />
  </BrowserRouter>,
);

describe('Register', () => {
  it('should render correctly', () => {
    const component = mount(
      <BrowserRouter>
        <Register {...props} />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with loader', () => {
    const component = mount(
      <BrowserRouter>
        <Register {...loadingProps} />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  it('mounts correctly', () => {
    const DEFAULT_STATE = {
      auth: {
        isLoading: false,
      },
    };
    const action = mapStateToProps(DEFAULT_STATE);
    expect(action).toEqual(DEFAULT_STATE.auth);
  });

  it('correctly initializes the value of the username field to an empty string', () => {
    expect(mountWrapper.find('input[name="username"]').prop('value')).toBe('');
    expect(mountWrapper.find('input[name="email"]').prop('value')).toBe('');
    expect(mountWrapper.find('input[name="password"]').prop('value')).toBe('');
    expect(mountWrapper.find('input[name="firstname"]').prop('value')).toBe('');
    expect(mountWrapper.find('input[name="lastname"]').prop('value')).toBe('');
    expect(mountWrapper.find('input[name="othername"]').prop('value')).toBe('');
    expect(mountWrapper.find('input[name="phonenumber"]').prop('value')).toBe(
      '',
    );
    expect(
      mountWrapper.find('input[name="confirmpassword"]').prop('value'),
    ).toBe('');
  });

  it('should submit the form', () => {
    const email = mountWrapper.find('input[name="email"]');
    email.instance().value = 'nedyudombat@gmail.com';
    email.simulate('change');
    const password = mountWrapper.find('input[name="password"]');
    password.instance().value = 'Iamtheadmin';
    password.simulate('change');
    const form = mountWrapper.find('form');
    form.simulate('submit', { preventDefault: jest.fn() });
    expect(mountWrapper.state()).toEqual(null);
  });
});
