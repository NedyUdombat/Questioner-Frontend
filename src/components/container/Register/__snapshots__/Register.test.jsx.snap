// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Register
      isLoading={false}
      register={[MockFunction]}
    >
      <div>
        <p
          className="f-50 text-center font-title"
          style={
            Object {
              "marginBottom": "0.625rem",
            }
          }
        >
          Register
        </p>
        <div
          className="auth-card"
        >
          <form
            autoComplete="on"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="firstname"
                  onChange={[Function]}
                  placeholder="First Name"
                  type="text"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="firstname"
                    onChange={[Function]}
                    placeholder="First Name"
                    type="text"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="lastname"
                  onChange={[Function]}
                  placeholder="Last Name"
                  type="text"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="lastname"
                    onChange={[Function]}
                    placeholder="Last Name"
                    type="text"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="othername"
                  onChange={[Function]}
                  placeholder="Other Name"
                  type="text"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="othername"
                    onChange={[Function]}
                    placeholder="Other Name"
                    type="text"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="username"
                  onChange={[Function]}
                  placeholder="Username"
                  required="required"
                  type="text"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="username"
                    onChange={[Function]}
                    placeholder="Username"
                    required="required"
                    type="text"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required="required"
                  type="email"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required="required"
                    type="email"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="phonenumber"
                  onChange={[Function]}
                  pattern="{+0-9}[13]"
                  placeholder="Phone Number"
                  type="tel"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="phonenumber"
                    onChange={[Function]}
                    pattern="{+0-9}[13]"
                    placeholder="Phone Number"
                    type="tel"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="password"
                  onChange={[Function]}
                  pattern="(?=.*[a-z]).{6,}"
                  placeholder="Password"
                  required="required"
                  title="Must be 6 characters or more and contain at least lowercase letter"
                  type="password"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="password"
                    onChange={[Function]}
                    pattern="(?=.*[a-z]).{6,}"
                    placeholder="Password"
                    required="required"
                    title="Must be 6 characters or more and contain at least lowercase letter"
                    type="password"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="confirmpassword"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  required="required"
                  type="password"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="confirmpassword"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    required="required"
                    type="password"
                    value=""
                  />
                </Input>
              </div>
            </div>
            <div
              style={
                Object {
                  "marginTop": ".9235rem",
                }
              }
            >
              <Button
                className="btn-dark"
                type="submit"
                value="GET STARTED"
              >
                <button
                  className="btn btn-dark"
                  type="submit"
                >
                  GET STARTED
                </button>
              </Button>
            </div>
            <p
              className="f-20 text-grey text-center"
              style={
                Object {
                  "marginTop": ".9235rem",
                }
              }
            >
              Already have an account? Log in
               
              <Link
                className="text-blue"
                to="/login"
              >
                <a
                  className="text-blue"
                  href="/login"
                  onClick={[Function]}
                >
                  here
                </a>
              </Link>
            </p>
          </form>
        </div>
      </div>
    </Register>
  </Router>
</BrowserRouter>
`;

exports[`Register should render correctly with loader 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Register
      isLoading={true}
      register={[MockFunction]}
    >
      <div>
        <p
          className="f-50 text-center font-title"
          style={
            Object {
              "marginBottom": "0.625rem",
            }
          }
        >
          Register
        </p>
        <div
          className="auth-card"
        >
          <form
            autoComplete="on"
            onSubmit={[Function]}
          >
            <Dimmer
              active={true}
            >
              <DimmerInner
                active={true}
              >
                <Ref
                  innerRef={
                    Object {
                      "current": <div
                        class="ui active transition visible dimmer"
                        style="display: flex;"
                      >
                        <div
                          class="content"
                        >
                          <div
                            class="ui large text loader"
                          >
                            Loading
                          </div>
                        </div>
                      </div>,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <div
                          class="ui active transition visible dimmer"
                          style="display: flex;"
                        >
                          <div
                            class="content"
                          >
                            <div
                              class="ui large text loader"
                            >
                              Loading
                            </div>
                          </div>
                        </div>,
                      }
                    }
                  >
                    <div
                      className="ui active transition visible dimmer"
                      onClick={[Function]}
                    >
                      <div
                        className="content"
                      >
                        <Loader
                          size="large"
                        >
                          <div
                            className="ui large text loader"
                          >
                            Loading
                          </div>
                        </Loader>
                      </div>
                    </div>
                  </RefFindNode>
                </Ref>
              </DimmerInner>
            </Dimmer>
            <div
              className="form-group"
            >
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="firstname"
                  onChange={[Function]}
                  placeholder="First Name"
                  type="text"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="firstname"
                    onChange={[Function]}
                    placeholder="First Name"
                    type="text"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="lastname"
                  onChange={[Function]}
                  placeholder="Last Name"
                  type="text"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="lastname"
                    onChange={[Function]}
                    placeholder="Last Name"
                    type="text"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="othername"
                  onChange={[Function]}
                  placeholder="Other Name"
                  type="text"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="othername"
                    onChange={[Function]}
                    placeholder="Other Name"
                    type="text"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="username"
                  onChange={[Function]}
                  placeholder="Username"
                  required="required"
                  type="text"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="username"
                    onChange={[Function]}
                    placeholder="Username"
                    required="required"
                    type="text"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required="required"
                  type="email"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required="required"
                    type="email"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="phonenumber"
                  onChange={[Function]}
                  pattern="{+0-9}[13]"
                  placeholder="Phone Number"
                  type="tel"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="phonenumber"
                    onChange={[Function]}
                    pattern="{+0-9}[13]"
                    placeholder="Phone Number"
                    type="tel"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="password"
                  onChange={[Function]}
                  pattern="(?=.*[a-z]).{6,}"
                  placeholder="Password"
                  required="required"
                  title="Must be 6 characters or more and contain at least lowercase letter"
                  type="password"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="password"
                    onChange={[Function]}
                    pattern="(?=.*[a-z]).{6,}"
                    placeholder="Password"
                    required="required"
                    title="Must be 6 characters or more and contain at least lowercase letter"
                    type="password"
                    value=""
                  />
                </Input>
              </div>
              <div
                className="input-div d-flex justify-content-center"
              >
                <Input
                  className="form-control-input"
                  name="confirmpassword"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  required="required"
                  type="password"
                  value=""
                >
                  <input
                    className="form-control-input"
                    name="confirmpassword"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    required="required"
                    type="password"
                    value=""
                  />
                </Input>
              </div>
            </div>
            <div
              style={
                Object {
                  "marginTop": ".9235rem",
                }
              }
            >
              <Button
                className="btn-dark"
                type="submit"
                value="GET STARTED"
              >
                <button
                  className="btn btn-dark"
                  type="submit"
                >
                  GET STARTED
                </button>
              </Button>
            </div>
            <p
              className="f-20 text-grey text-center"
              style={
                Object {
                  "marginTop": ".9235rem",
                }
              }
            >
              Already have an account? Log in
               
              <Link
                className="text-blue"
                to="/login"
              >
                <a
                  className="text-blue"
                  href="/login"
                  onClick={[Function]}
                >
                  here
                </a>
              </Link>
            </p>
          </form>
        </div>
      </div>
    </Register>
  </Router>
</BrowserRouter>
`;
